src/openvic-simulation/scripts/ConditionalWeight.cpp

```diff
@@ -26,23 +26,76 @@ static NodeCallback auto expect_modifier(
 }
 
 node_callback_t ConditionalWeight::expect_conditional_weight(base_key_t base_key) {
```

Comment:
Why not split this up in a method per base_key_t.
I don't see much overlap.

Reply:
The `modifier` and `group` cases in the `expect_dictionary_key_map` call below are needed for all base types, and in general I prefer to generalise functions where possible.